### ISSUE TITLE
WARNING: Cannot load code editor, Ace library not loaded. Falling back to plain text editor

### ISSUE THREAD

**Issue by bstienst** [josdejong/jsoneditor#176]
Using 4.1.3 dist with the following code:

	<link href="js/jsoneditor/jsoneditor.css" rel="stylesheet" type="text/css">
	<script src="js/jsoneditor/jsoneditor.js"></script>

...							

							var container = document.getElementById('jsoneditor');
							  
							var options = {
								mode: 'tree',
								modes: ['code', 'form', 'text', 'tree', 'view'], // allowed modes
								error: function (err) {
								  alert(err.toString());
								}
							  };

							jsoneditor = new JSONEditor(container, options, json);

when I switch to code mode, I get the following error:

WARNING: Cannot load code editor, Ace library not loaded. Falling back to plain text editor

I've tried Chrome and Firefox.

**Comment by josdejong** [comment]
Thanks for letting me know. 

Do the examples that come with the library work for you? And do you possibly have an other Ace editor loaded in the same web application? (there are a few open issues related to that)

**Comment by bstienst** [comment]
Yes the examples work.  No I don't have another Ace editor on my page.  

I am using Bootstrap and Jquery.  Is there anything you can suggest that might help debug the problem?   I have very little knowledge of the ace editor and the techniques you are using to load it.

**Comment by josdejong** [comment]
Is it possible for you to create a [jsbin](http://jsbin.com) demonstrating the issue? Or can you post your code? (stripped down as much as possible)

**Comment by bstienst** [comment]
I certainly can.  Can you point me to where jsoneditor.css and jsoneditor.js are available online?  All the other included js files I can find elsewhere.

**Comment by josdejong** [comment]
Ok great. You can use the files via cdnjs:

https://cdnjs.com/libraries/jsoneditor

https://cdnjs.cloudflare.com/ajax/libs/jsoneditor/4.1.3/jsoneditor.js
https://cdnjs.cloudflare.com/ajax/libs/jsoneditor/4.1.3/jsoneditor.css

(or the minified ones)

**Comment by andrewboni** [comment]
+1 on this

**Comment by andrewboni** [comment]
I'm using requirejs to load both ace and jsoneditor. Using `code` mode, I get this error:

```
WARNING: Cannot load code editor, Ace library not loaded. Falling back to plain text editor
```

**Comment by andrewboni** [comment]
Ace is loaded on the page, however

**Comment by josdejong** [comment]
Yes, it's probably this combination of having ace loaded yourself *and* loading JSONEditor. Will fix it asap.

**Comment by josdejong** [comment]
I've just released v4.2.0, which has a new option `ace`, allowing you to pass a custom instance of Ace instead of the embedded one. This may solve your issue.
